### PR TITLE
feat: add Caltech101 Silouettes dataset

### PR DESCRIPTION
**Pull request by meakio**
Hi,

I created a loader for Caltech101 dataset inspired by Mnist dataset. The data are downloaded from [https://people.cs.umass.edu/~marlin/data.shtml](https://people.cs.umass.edu/~marlin/data.shtml).

Because you used only train or test split, I merged train and validation set in train.

There are 2 problems. First, I used scipy.io.loadmat to load the file and it add scipy library as a dependancy if we want to use this dataset. Secondly, I did not check the file like you have done with the magic number for Mnist.

Feel free to comment and propose improvement, I will update this pull request accordingly.

### CODE REVIEW

File: tensorpack/libinfo.py [hunk 1]
```diff
@@ -67,4 +67,4 @@
 # These lines will be programatically read/write by setup.py
 # Don't touch them.
 __version__ = '0.9.4'
-__git_version__ = __version__
+__git_version__ = "v0.9.4-22-g8d888d30-dirty"
```
Comment (ppwwyyxx) [inline tensorpack/libinfo.py]: Please undo this.

File: tensorpack/dataflow/dataset/caltech101.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    if not os.path.exists(filepath):
+        logger.info("Downloading to {}...".format(filepath))
+        download(url, work_directory)
+    return filepath
+
+
+class Caltech101(RNGDataFlow):
+    """
+    Produces [image, label] in Caltech101 dataset,
+    image is 28x28 in the range [0,1], label is an int.
```
Comment (ppwwyyxx) [inline tensorpack/dataflow/dataset/caltech101.py]: please specify the range of the int

**Comment by meakio** [comment]
I rename it to Caltech101Silhouettes and split it in train, val and test as in BSDS500.
I also undo adding scipy in requirements.

File: tensorpack/libinfo.py (hunk 1 above)
Comment (meakio) [reply tensorpack/libinfo.py]: Done

File: tensorpack/dataflow/dataset/caltech101.py (hunk 2 above)
Comment (meakio) [reply tensorpack/dataflow/dataset/caltech101.py]: Done

**Comment by ppwwyyxx** [comment]
Thanks!

**Pull request merged**